### PR TITLE
Set `ResizableEditor` height based on border-box

### LINKED ISSUE #66297: Resizable Editor: Unintended vertical scrollbar
### Description

In the resizable editor, if there is little content, the vertical scrollbar is not necessary, but it seems to always be displayed.

After investigating this issue, I found that it was caused by the top border added by #66041.

### Step-by-step reproduction instructions

- Open the site editor.
- Open the pattern or template part editor.

### Screenshots, screen recording, code snippet

Image

If I temporarily disable the top border this issue doesn't occur:

Image

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### PR DESCRIPTION
**Pull request by stokesman**
## What?
Updates usage of `useResizeObserver`.

## Why?
To remove an element that doesn’t have to be rendered. Also to size the canvas in what should be a more robust way—by its border box, thereby avoiding a corner case that a theme applies a border on `body` which would cause overflow if sizing based on the content box.

## How?
- Uses the latest API of the resize observer hook which eliminates having to render a sizing element.
- Uses the reported border box size to set the the height of the resizable editor.

## Testing Instructions
1. For better edge case detection, set your OS to always display scrollbars, if it doesn't already
2. Open up the site editor
3. Go to the Patterns editor and select or create a pattern to open the editor view that includes the resizable editor
4. Check for any visual issues

### CODE REVIEW

File: packages/editor/src/components/visual-editor/index.js [hunk 1]
```diff
@@ -323,21 +326,6 @@ function VisualEditor( {
...
-		renderingMode === 'post-only' ? typewriterRef : null,
-		useFlashEditableBlocks( {
-			isEnabled: renderingMode === 'template-locked',
-		} ),
-		useSelectNearestEditableBlock( {
-			isEnabled: renderingMode === 'template-locked',
-		} ),
-		useZoomOutModeExit(),
-	] );
-
```
Comment (stokesman) [inline packages/editor/src/components/visual-editor/index.js:340]: Just moved these down some lines so they’d be after the `enableResizing` declaration as the arguments to `useMergeRefs` now depends on it.

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

As I understand it, removing the transparent border via #66390 will eventually fix the issue this PR is trying to solve, right?

However, since moving to non-legacy hooks is a good idea, how about removing the backport label from this PR and moving it forward separately from #66390?

**Review by t-hamano — APPROVED** [review]
This PR is working as expected. With #66390 merged, this PR will not cause any visual changes, but removing the invisible div may prevent some future issues.

| Before| After |
|--------|--------|
| before | after |

**Comment by stokesman** [comment]
Thanks for testing and reviewing Aki!

**Pull request merged**